src/java/org/apache/cassandra/tools/nodetool/stats/DataPathsPrinter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.tools.nodetool.stats;
+
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Map;
+
+public class DataPathsPrinter<T extends StatsHolder>
+{
+    public static <T extends StatsHolder> StatsPrinter<T> from (String format)
+    {
+        switch(format)
+        {
+            case "json":
+                return new StatsPrinter.JsonPrinter<T>();
+            case "yaml":
+                return new StatsPrinter.YamlPrinter<T>();
+            default:
+                return (StatsPrinter<T>) new DefaultPrinter();
+        }
+    }
+    
+    public static class DefaultPrinter implements StatsPrinter<DataPathsHolder>
+    {
+
+        @Override
+        public void print(DataPathsHolder data, PrintStream out)
+        {
+            for (String keyspace : data.pathsHash.keySet())
```

Comment:
my IDE informs me that this might be reworked with

            for (Map.Entry<String, Object> entry : data.pathsHash.entrySet())
            {
                out.println("Keyspace : " + entry.getKey());
                Map<String, List<String>> ksPaths = (Map<String, List<String>>) entry.getValue();

You would work with entry instead of getting keyspaces as keys and then getting its entry via "get" from the map again. That is not necessary. It is better to work with isolated entry instead only.

Reply:
or maybe not huh, this is complicated / confusing, i need to go over this again.

Reply:
Yes, the ultimate version of this is like this:

            for (Map.Entry<String, Object> entry : data.pathsHash.entrySet())
            {
                out.println("Keyspace : " + entry.getKey());
                Map<String, List<String>> ksPaths = (Map<String, List<String>>) entry.getValue();
                for (Map.Entry<String, List<String>> table : ksPaths.entrySet())
                {
                    out.println("\tTable : " + table.getKey());
                    out.println("\tPaths :");
                    for (String path : table.getValue())

Reply:
@rtib I do not like the fact that there is "------------" at the end of the output when there is not other table to print. You would need to rewrite this to solution using Iterator and you would ask if you are going to print "-----" only if iterator has next entry, like this:

        public void print(DataPathsHolder data, PrintStream out)
        {
            Iterator<Map.Entry<String, Object>> iterator = data.pathsHash.entrySet().iterator();

            while (iterator.hasNext())
            {
                Map.Entry<String, Object> entry = iterator.next();

                out.println("Keyspace : " + entry.getKey());
                Map<String, List<String>> ksPaths = (Map<String, List<String>>) entry.getValue();
                for (Map.Entry<String, List<String>> table : ksPaths.entrySet())
                {
                    out.println("\tTable : " + table.getKey());
                    out.println("\tPaths :");
                    for (String path : table.getValue())
                    {
                        out.println("\t\t" + path);
                    }
                    out.println("");
                }

                if (iterator.hasNext())
                {
                    out.println("----------------");
                }
            }
        }

Reply:
Oh, I was also stumbling over that one, however, I found the readability of having a loop variable of type `String` some better than `Map.Entry<String, List<String>>`. What if solving the separator problem otherwise?

Reply:
I prefer the iterator approach and working with one entry only at the time. Having keys in for loop and then performing get on that map again is not too much performance friendly. It also solves not printing the separator at the very end of the output by looking if hasNext returns true.